Bullwinkle.class.nut

```diff
@@ -385,7 +395,7 @@ class Bullwinkle {
 
             // Add the retry information
             message.retry <- {
-                "ts": time() + timeout,
+                "ts": ( Bullwinkle._isAgent() ? time() : hardware.micros()/1000000 )  + timeout,
```

Comment:
Generally, the send ts and the eventTime are the same - its only when you introduce SPIFlash/retries that things get a bit hairy...

In my head, the .ts was always the eventTime, and it was just never properly implemented (the implementation and comments around the ```_messageFactory``` function seem to support this).  

Ultimately, it is just a semantics thing - I would just argue that the message.ts is most useful if it is the eventTime and not the sending time.  The Bullwinkle.Package is what calculates things like latencies anyway (since you need the same timebase to calculate accurately) so I'm not sure how the message.ts as the sending time is more useful than as the eventTime?  

Regardless, it does need to be clearly defined in the docs (instead of ambiguous as it is now ```Timestamp when message was created``` in Bullwinkle? or originally in the HW?) and implemented to match.  I think this implementation is more useful and shouldn't be breaking to anyone (the default behavior is to autogenerate timestamps each time the message is sent, just as it is now) , however, if you can come up with some reasons that line of thinking is wrong, I'll be happy to relent :)